### ISSUE TITLE
Docs should include info how to promisify jar

### ISSUE THREAD

**Issue by thamerla** [salesforce/tough-cookie#143]
I would like to use promises but I have no clue how to promisify methods using built-in util.promisify. Docs should support this if it's not currently possible to implement promises on their own.

**Comment by stash** [comment]
I suspect that `util.promisfy` should Just Work (tm) but I'll write up some test cases and update README.md to reflect the preferred method. 

Since we support minimum node.js v6, and `util.promisify` was added in v8, tough-cookie unfortunately can't (yet!) switch to supporting this automatically. So, I'd like to explore a wrapper package that offers a promisify-ed API.

**Comment by stash** [comment]
See also: #66

**Comment by nonyy** [comment]
@stash I saw that one and it prompted me to write this issue.

**Comment by bshevlin** [comment]
@stash [Bluebird](http://bluebirdjs.com/docs/api-reference.html) has a good backwards-compatible promisification function and would probably help. They also have util functions that could help with backwards compatibility.

[An example from Bluebird's docs](http://bluebirdjs.com/docs/api/ascallback.html):
```
function getDataFor(input, callback) {
    return dataFromDataBase(input).asCallback(callback);
}
```

**Comment by shivam-tripathi** [comment]
This should be useful:
```
const cookieJar = new tough.CookieJar();
const setCookie = util.promisify(cookieJar.setCookie.bind(cookieJar));
await setCookie('billi=0', 'https://www.catcat.com/', {
    ignoreError: false
});
```

**Comment by awaterma** [comment]
We now support promises in the 4.x release. I think this issue is asking for better documentation, and some examples? Or should we close this out?

**Comment by colincasey** [comment]
this should be addressed in #234

**Comment by wjhsf** [comment]
#234 has been merged with updates to the README.
